LayoutTests/requestidlecallback/requestidlecallback-does-not-block-timer.html

```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html><!-- webkit-test-runner [ RequestIdleCallbackEnabled=true ] -->
+<html>
+<body>
+<script src="../resources/js-test.js"></script>
+<script>
+
+description('This tests that scheduling an idle callback would not delay the firing of a timer.');
+
+jsTestIsAsync = true;
+
+function synchronouslyWait(milliseconds) {
+    let start = performance.now();
+    while (performance.now() - start < milliseconds);
+}
+
+let didRunIdleCallback = false;
+onload = () => {
+    requestIdleCallback((idleDeadline) => {
+        shouldBeTrue('didFireTimer');
+        window.idleDeadline = idleDeadline;
+        shouldBeTrue('idleDeadline.timeRemaining() > 0');
+        synchronouslyWait(50);
+        shouldBe('idleDeadline.timeRemaining()', '0');
+        didRunIdleCallback = true;
+    });
+    setTimeout(() => {
+        didFireTimer = true;
+    }, 10);
```

Comment:
Looks like 10ms is too long of a delay. The test is flaky. Will change it to 0ms.